File: newsfeedsdk/src/main/java/com/appyhigh/newsfeedsdk/encryption/LogDetail.java
package com.appyhigh.newsfeedsdk.encryption;


import android.content.Context;
import android.util.Log;
import android.widget.Toast;

import com.appyhigh.newsfeedsdk.BuildConfig;


public class LogDetail {

    private static boolean debugger = false;

    public static void LogD(String T, String S) {
        if (debugger) {
            if (S != null) {
                Log.d(T, S);
            }
        }
    }
    public static void LogDE(String T, String S) {
        if (debugger) {
            if (S != null) {
                Log.e(T, S);
            }
        }
    }

    public static void ToastD(Context T, String S) {
        if (debugger) {
            Toast.makeText(T, S, Toast.LENGTH_SHORT).show();
        }
    }

    // Release mode code
    public static void LogR(String T, String S) {
        if (S != null) {
            Log.d(T, S);
        }
    }

    public static void ToastR(Context T, String S) {
        Toast.makeText(T, S, Toast.LENGTH_SHORT).show();
    }

    public static void println(int ss) {
        println(String.valueOf(ss));
    }

    public static void println(String ss) {
        System.out.println(ss);

        /*if (debugger) {
            System.out.println(ss);
        }*/
    }
}

File: newsfeedsdk/src/main/java/com/appyhigh/newsfeedsdk/encryption/LogDetail.kt
package com.appyhigh.newsfeedsdk.encryption

import android.content.Context
import android.content.pm.ApplicationInfo
import android.content.pm.PackageManager
import android.util.Log
import android.widget.Toast
import com.appyhigh.newsfeedsdk.Constants

object LogDetail {
    private var debugger = true
    
//    @JvmStatic
//    fun init(context: Context){
//        try{
//            val ai: ApplicationInfo = context.packageManager.getApplicationInfo(context.packageName, PackageManager.GET_META_DATA)
//            val bundle = ai.metaData
//            debugger = bundle.getBoolean(Constants.FEED_DEBUGGER, false)
//        } catch (ex:Exception){ }
//    }
    @JvmStatic
    fun init(){}
    
    @JvmStatic
    fun LogD(T: String?, S: String?) {
        if (debugger) {
            if (S != null) {
                Log.d(T, S)
            }
        }
    }

    @JvmStatic
    fun LogDE(T: String?, S: String?) {
        if (debugger) {
            if (S != null) {
                Log.e(T, S)
            }
        }
    }

    @JvmStatic
    fun LogDE(T: String?, S: String?, e:Exception?) {
        if (debugger) {
            if (S != null) {
                Log.e(T, S)
            }
            e?.printStackTrace()
        }
    }

    @JvmStatic
    fun LogEStack(e:Exception?) {
        if (debugger) {
            e?.printStackTrace()
        }
    }

    @JvmStatic
    fun LogEStack(e:Throwable?) {
        if (debugger) {
            e?.printStackTrace()
        }
    }

    fun ToastD(T: Context?, S: String?) {
        if (debugger) {
            Toast.makeText(T, S, Toast.LENGTH_SHORT).show()
        }
    }

    // Release mode code
    fun LogR(T: String?, S: String?) {
        if (S != null) {
            Log.d(T, S)
        }
    }

    fun ToastR(T: Context?, S: String?) {
        Toast.makeText(T, S, Toast.LENGTH_SHORT).show()
    }

    fun println(ss: Int) {
        println(ss.toString())
    }

    @JvmStatic
    fun println(ss: String?) {
        try{
//            println(ss)
            /*if (debugger) {
                LogDetail.LogDE(ss);
            }*/
        } catch (ex:Exception){
            LogDetail.LogEStack(ex)
        }
    }
}